srecord/output/file/aomf.cc

```diff
@@ -85,7 +86,7 @@ void
 srecord::output_file_aomf::content_record(unsigned long address,
     const unsigned char *data, size_t len)
 {
-    size_t maxlen = 4 * srecord::record::max_data_length;
+    size_t maxlen = static_cast<size_t>(4 * srecord::record::)max_data_length;
```

Comment:
I'm sure the code is potentially less error prone however given the mental gymnastics required to parse this one's head is more susceptible to bugs in other ways. But I can't deny the better result.

> Programs must be written for people to read, and only incidentally for machines to execute. ― Harold Abelson

Unless you feel strongly about this one, I'd rather leave it out. Perhaps there are more straight forward ways to address the underlying potential bug found than the suggested?

Reply:
As per comments elsewhere, the warning is likely valid and we should fix it. The auto-correct suggestions may not be the optimal/cleanest fix though. I'd like to leave the check in at a minimum

Reply:
This is one that has to be manually checked:
`max_data_length` is an enum, with value 255. I don't see a reason why it's not declared as const uint8.
And if we declare a const uint8 for the magic value 4, the result will fit into size_t without cast.

I think all multiplication have to be checked, if there can be an overflow. Better than static_cast is to do the multiplication with a type big enough, and check the result (at least with assert).

Reply:
Agreed on all points and I do prefer the option of using a large enough type. 

Also, magic numbers make me cringe. But definitely not wanting all of them to be fixed in this PR.